velero-plugins/imagestream/shared_test.go

```diff
@@ -0,0 +1,427 @@
+package imagestream
+
+import (
+	"context"
+	"slices"
+	"testing"
+
+	"github.com/konveyor/openshift-velero-plugin/velero-plugins/clients"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/restmapper"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/yaml"
+)
+
+func TestGetRegistryEnvsForLocation(t *testing.T) {
+	type args struct {
+		location  string
+		namespace string
+	}
+	tests := []struct {
+		name          string
+		args          args
+		objs          []runtime.Object
+		want          []string
+		wantErr       bool
+		wantErrString string
+	}{
+		{
+			name: "simple aws case, operator created registry secret",
+			args: args{
+				location:  "bsl1",
+				namespace: "ns1",
+			},
+			objs: []runtime.Object{
+				&corev1.Namespace{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Namespace",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "ns1",
+					},
+				},
+				&corev1.Secret{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Secret",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "oadp-bsl1-aws-registry-secret",
+						Namespace: "ns1",
+					},
+					Data: map[string][]byte{
+						"access_key": []byte("ak"),
+						"secret_key": []byte("sk"),
+					},
+				},
+				&velerov1.BackupStorageLocation{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "BackupStorageLocation",
+						APIVersion: velerov1.SchemeGroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "bsl1",
+						Namespace: "ns1",
+					},
+					Spec: velerov1.BackupStorageLocationSpec{
+						Provider: "aws",
+						Default:  true,
+						StorageType: velerov1.StorageType{
+							ObjectStorage: &velerov1.ObjectStorageLocation{
+								Bucket: "buc",
+								Prefix: "prefix",
+							},
+						},
+						Credential: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "cloud-credentials",
+							},
+							Key: "cloud",
+						},
+					},
+				},
+			},
+			want: []string{
+				"REGISTRY_STORAGE=s3",
+				"REGISTRY_STORAGE_S3_BUCKET=buc",
+				"REGISTRY_STORAGE_S3_REGION=us-east-2",
```

Comment:
probably someone's bucket name that is called `buc` is in us-east-2 since we have an auto region discovery feature.